sample_metadata/parser/generic_parser.py

```diff
@@ -305,6 +305,18 @@ async def get_cpg_sample_id(self, row: SingleRow) -> Optional[str]:
     def get_participant_id(self, row: SingleRow) -> Optional[str]:
         """Get external participant ID from row"""
 
+    @abstractmethod
+    def get_reported_sex(self, row: GroupedRow) -> Optional[int]:
+        """Get reported sex from grouped row"""
+
+    @abstractmethod
+    def get_reported_gender(self, row: GroupedRow) -> Optional[str]:
+        """Get reported gender from grouped row"""
+
+    @abstractmethod
+    def get_karyotype(self, row: GroupedRow) -> Optional[str]:
+        """Get karyotype from grouped row"""
```

Comment:
Marking these as abstract requires the subclass to implement it, I think a default implementation of returning None should be fine. That will also stop the linting errors.